docs: how-arbitrum-works-reorg (#1866)
OffchainLabs/arbitrum-docs: arbitrum-docs/how-arbitrum-works/02-transaction-lifecycle.mdx

---

> title: Sequencing, Followed by Deterministic Execution

Reviewer: Sidebar entry is inconsistent with the new title (they are different)

Author: updated

---
title: Transaction Lifecycle

> ---

Reviewer 2: This should probably be a complete front matter.

Author: Adding front matter for all 👍

> ## Sequencing, Followed by Deterministic Execution

Reviewer: This could be the section of the page, to align with the 4 big ideas mentioned in the Introduction.
This first part serves as a "In a nutshell" (a big nutshell though) section to the rest of the page, since it's summarizing the lifecycle of a transaction.

The second part of the page goes into much more detail than this, so as a reader, I'd like to know that this bit is going to introduce me to what's going to be explained more in detail afterwards.

Author: I'll rename the page to "Sequencing..." 

As to separate the first and second halves of the doc, currently the second section is "Overview: The lifecycle..." since this is a deeper technical breakdown, I'm going to rename this to: "Deep dive: The lifecycle" to delineate the "summary/top" from the "technical/bottom"

The state transition function is deterministic, which means that its behavior depends only on the current state and the contents of the next transaction--and nothing else. Because of this determinism, the result of a transaction T will depend only on the genesis state of the chain, the transactions before T in the sequence, and T itself.

It follows that anyone who knows the transaction sequence can compute the state transition function for themselves--and all honest parties who do this are guaranteed to get identical results. This is the normal way that Nitro nodes operate: get the transaction sequence, and run the state transition function locally. No consensus mechanism is needed for this.

> ## Overview: The Lifecycle of an Arbitrum Transaction

Reviewer: ```suggestion
## The lifecycle of an Arbitrum transaction
```
The section above is more of an overview than this one. This section goes into further details, imo.

Author: renamed - see previous comment

> As an introduction to the various components that compose the Arbitrum protocol, we'll go step-by-step over the phases an Arbitrum transaction goes through, starting with a client creating a signed transaction, to it ultimately being confirmed back on layer 1.

Reviewer: ```suggestion
We'll now go step-by-step over the phases an Arbitrum transaction goes through, starting with a client creating a signed transaction, to it ultimately being confirmed back on layer 1.
```
This is no longer an introduction. We can change this a bit to follow that logic.

> As an introduction to the various components that compose the Arbitrum protocol, we'll go step-by-step over the phases an Arbitrum transaction goes through, starting with a client creating a signed transaction, to it ultimately being confirmed back on layer 1.

Reviewer: Two more things about this section:
- The introduction to the section might need some tweaks in the wording. There's a mention to "This overview..." further down.
- The section is full of broken links. Each link should be tested.

Author: Addressing phrasing. A few things are a little confusing like referring to phases but this will be resolved in the next iteration. Will test links 👍

**See:**

- [Challenges](/how-arbitrum-works/07-interactive-fraud-proofs.mdx#challengemanager)
- [Wasm/WAVM](/how-arbitrum-works/07-interactive-fraud-proofs.mdx#wasm-to-wavm)

> L1 contracts also keep track of the tree of all assertions; i.e., how many stakers are in disagreement, who is currently disputing with whom, etc. We refer to this level of Arbitrum's design architecture as its "assertion tree protocol."

Reviewer: The "Assertion Tree Protocol" just below this paragraph is pointing at the page in the old structure, which doesn't have a sidebar.
We should move all these pages to a separate folder (for example `website/archive`), or remove them.

Author: will archive, update links
